gef.py

```diff
@@ -1027,7 +1026,7 @@ def get_next(self):
         return GlibcArena("*{:#x} ".format(addr_next))
 
     def is_main_arena(self):
-        return int(self) == parse_address("&main_arena")
+        return int(self) == search_for_main_arena()
```

Comment:
Good call!
Unfortunately the `search_for_main_arena` function currently has the side effect of changing the global state by setting `__gef_current_arena__`. So at the moment this fix would always modify said global variable when `is_main_arena` is called (currently, in `heap_addr()`, `get_heap_info_list()` and `GlibcHeapChunksCommand.dump_chunks_arena()`) and could very likely introduce new bugs. To fix this the `search_for_main_arena` function should probably be modified to not change the global variable, but to do so in the calling code everywhere this is necessary.

Reply:
Agreed, it will become clunky especially for other part of the code like `get_glibc_arena()` - actually re-reading that code I'm not sure why this function is necessary at all since `GlibcArena` does just that.
For now, I suggest we can just validate this for fixing this issue, and add the libc heap classes to be reviewed for the big code refactor coming in.

Reply:
Let's not add something that changes state to a predicate function. Maybe we split `search_for_main_arena` by adding a new function like `search_for_and_set_main_arena` (not actually that name). That way we can *just* search for it in this function.